Bottom: baf3f524060a12357e217dbe0819e40fc524ae6b
Top:    557b7d4a8b3c04b588251a4b0e1f40e499a60dd5
Author: Peter Clifton <pcjc2@cam.ac.uk>
Date:   2011-04-25 15:58:27 +0100

Remove hack in draw.c which broke PS (and probably other) output

The set_layer call hacked into DrawLayerGroup to reset compositing tricked
the PS exporter to re-open and overwrite output in the main layer output
file during the fab drawing output.


---

diff --git a/src/draw.c b/src/draw.c
index fa3d176..ac5248b 100644
--- a/src/draw.c
+++ b/src/draw.c
@@ -1089,10 +1089,6 @@ DrawLayerCommon (LayerTypePtr Layer, const BoxType * screen, bool clear_pins)
   clip_box = screen;
   r_search (Layer->polygon_tree, screen, NULL, poly_callback, Layer);
 
-  /* HACK: Subcomposite polygons separately from other layer primitives */
-  /* Reset the compositing */
-  gui->set_layer (0, GetLayerGroupNumberByPointer (Layer), 0);
-
   if (clear_pins && TEST_FLAG (CHECKPLANESFLAG, PCB))
     return;
 
@@ -1142,8 +1138,7 @@ DrawLayerGroup (int group, const BoxType *drawn_area)
   Cardinal *layers = PCB->LayerGroups.Entries[group];
 
   clip_box = drawn_area;
-  for (i = n_entries - 1; i >= 0;
-      i--, gui->set_layer (0, group, 0)) /* HACK: Subcomposite each layer in a layer group separately */
+  for (i = n_entries - 1; i >= 0; i--)
     {
       layernum = layers[i];
       Layer = PCB->Data->Layer + layers[i];
